asterisk/agi/src/Dialplan/Retails.php

```diff
@@ -98,9 +104,17 @@ public function process()
         // Some feedback for asterisk cli
         $this->agi->notice("Processing outgoing call from \e[0;36m%s\e[0;93m to number %s", $retailAccount, $exten);
 
-        // All retail account calls are handled as external
-        $this->externalNumberAction
-            ->setDestination($exten)
-            ->process();
+        $cfw_destination= $this->agi->getSIPHeader('X-Info-Cfw-Destination');
+        if ($cfw_destination) {
+            $dstRetailAccount = $this->endpointResolver->getRetailFromEndpoint($cfw_destination);
+            $this->retailCallAction
+                ->setRetailAccount($dstRetailAccount)
+                ->process();
+        } else {
+            // Remaining retail account calls are handled as external
+            $this->externalNumberAction
+                ->setDestination($exten)
+                ->process();
+        }
```

Comment:
Another possible approach here is passing CallForwardSettingId in a header and using CallForwardAction instead of these two actions. In that case, CallForwardAction would require some adaptations because it only supports Users/Residential forwards.


Example (require EntityManagerInterface and CallForwardAction in constructor) :

```
        $cfwSettingId = $this->agi->getSIPHeader('X-Info-Cfw-Id');

        /** @var CallForwardSettingsRepository $cfwRepository */
        $cfwRepository = $this->em->getRepository(CallForwardSetting::class);

        /** @var CallForwardSetting|null $cfwSetting */
        $cfwSetting = $cfwRepository->find($cfwSettingId);
 
        // Process Retail Call Forward 
        $this->callForwardAction
            ->setCallForward($cfwSetting)
            ->process();